luaui/Widgets/api_unit_swap.lua

```diff
@@ -0,0 +1,305 @@
+function widget:GetInfo()
+	return {
+		name = "Unit Swap API",
+		desc = "-",
+		license = "GNU GPL, v2 or later",
+		layer = -1,
+		enabled = true
+	}
+end
+
+---Creates a simple enum-like table, where for each entry, the key and value are the same. This allows syntax like
+---ENUM.OPTION_ONE, while also having the value "OPTION_ONE" for serialization or printing.
+---@param ... table a list of entries for the enum
+---@return table
+local function enum(...)
+	local args = { ... }
+	local result = {}
+	for _, v in ipairs(args) do
+		result[v] = v
+	end
+	return result
+end
+
+local factions = enum("arm", "cor", "leg")
+local terrains = enum("ground", "water")
+
+local function isFactionEnabled(faction)
+	if faction == factions.leg then
+		return Spring.GetModOptions().experimentallegionfaction or false
+	else
+		return true
+	end
+end
+
+local factionSwapNames = {
```

Comment:
Consider making this file strictly an API. This means:
* `factions`, `terrains`, `factionSwapNames`, and `terrainSwapNames` become empty tables here.
* add `WG.UnitSwap.RegisterFaction(string: factionName, bool or function returning a bool: enabled)`
* add `WG.UnitSwap.RegisterTerrain(string: terrainName)`.
* add `WG.UnitSwap.Register{Terrain,Faction}SwapNames(table)`.
* a separate widget, `default_unit_swaps.lua` or something, just passes the existing hardcoded tables.

This way, when in the future there are mods that add units and want to modify the swap relationships, or if players want to add something custom, nobody needs to override this core widget, which avoids obsolete code spreading and sticking around.

Reply:
Oh I like that idea a lot!